bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/HueBridge.java

```diff
@@ -1129,4 +984,117 @@ private String getRelativeURL(String path) {
         }
         return path.isEmpty() ? relativeUrl : relativeUrl + "/" + path;
     }
+
+    public HueResult get(String address) throws CommunicationException {
+        return doNetwork(address, HttpMethod.GET);
+    }
+
+    public HueResult post(String address, String body) throws CommunicationException {
+        return doNetwork(address, HttpMethod.POST, body);
+    }
+
+    public HueResult put(String address, String body) throws CommunicationException {
+        return doNetwork(address, HttpMethod.PUT, body);
+    }
+
+    public HueResult delete(String address) throws CommunicationException {
+        return doNetwork(address, HttpMethod.DELETE);
+    }
+
+    private HueResult doNetwork(String address, HttpMethod requestMethod) throws CommunicationException {
+        return doNetwork(address, requestMethod, null);
+    }
+
+    private HueResult doNetwork(String address, HttpMethod requestMethod, @Nullable String body)
+            throws CommunicationException {
+        logger.trace("Hue request: {} - URL = '{}'", requestMethod, address);
+        try {
+            final Request request = httpClient.newRequest(address).method(requestMethod).timeout(timeout,
+                    TimeUnit.SECONDS);
+
+            if (body != null) {
+                logger.trace("Hue request body: '{}'", body);
+                request.content(new StringContentProvider(body), "application/json");
+            }
+
+            final ContentResponse contentResponse = request.send();
+
+            final int httpStatus = contentResponse.getStatus();
+            final String content = contentResponse.getContentAsString();
+            logger.trace("Hue response: status = {}, content = '{}'", httpStatus, content);
+            return new HueResult(content, httpStatus);
+        } catch (ExecutionException | InterruptedException | TimeoutException e) {
```

Comment:
Please execute `Thread.currentThread().interrupt();` in case of InterruptedException before logging and throwing `CommunicationException`